### PR TITLE
Modern Theme

### PR DESCRIPTION
**Pull request by crhallberg**
DEMO: http://nero.library.villanova.edu/bleedingedge/

- [x] Waiting for the [Bootstrap Reduce PR](https://github.com/vufind-org/vufind/pull/958)
    - [x] Move Bootstrap changes over to PR
- [x] Create and expose more variables
- [x] Advanced search remove wrapping
- [x] Fix styling of tabbed search box (see screen shot in comments)
- [x] Fix broken "visual" view (also affects bootstrap3 base theme)
- [x] Fix minor inconsistency: tabs below search most of the time, but above after an advanced search has been applied.
- [x] Fix various issues sent in email from Demian to Chris on 7/3/17.
- [x] Fix styling of system maintenance message triggered by NoILS driver (affects all themes; see screen shot in comments).
- [x] Fix missing arrows in hierarchical facets (also affects bootstrap3 base theme)
- [x] Fix interaction of advanced search and tabs (see screen shot in comments)
- [x] Fix glitchy caret in "remove filters" box (see screen shot in comments)
- [x] Fix display problems for selected facets (see screen shot in comments)
- [x] Fix glitchy display of facet values in advanced search screen.
- [x] Rename modern to sandal.
    - [x] Rename folder
    - [x] Gruntfile
    - [x] [Image path](https://github.com/vufind-org/vufind/pull/961/files#diff-8ec87b9ebdacab68a375fad843aed4ccR2)
- [x] Port and test SASS

The following have been moved to 4.1
- [ ] Modern checkboxes (see https://github.com/crhallberg/vufind/tree/modern-checkboxes)
- [ ] Header float
- [ ] Advanced search float buttons

### CODE REVIEW

**Comment by black23** [comment]
In Czech language small visual bug at Advance Search:
http://prntscr.com/fm12pg

**Comment by crhallberg** [comment]
Thank you! I'll take care of it!

**Comment by black23** [comment]
I don't know that is some UX rule but I pefer group of buttons like this at Advaced Search
http://prntscr.com/fm1d85
If form is too long, maybe it would be better practice let buttons to flow instead of doubled them. See http://prntscr.com/fm1g5r

**Comment by crhallberg** [comment]
Adding some flow/floating elements is on the slate, this seems like a good place to try it out.

File: package.json [hunk 1]
```diff
@@ -11,9 +11,9 @@
   },
   "dependencies": {
     "grunt": "^0.4.5",
-    "grunt-sass": "^1.0.0",
-    "grunt-contrib-less": "1.3.0",
+    "grunt-contrib-less": "^1.3.0",
```
Comment (demiankatz) [inline package.json]: Is this just an alphabetical order thing for readability? If so, might as well commit it to master to reduce complexity of this PR.

File: themes/bootstrap3/templates/Recommend/TopFacets.phtml [hunk 2]
```diff
@@ -49,4 +49,4 @@
       <? endif; ?>
     </div>
   <? endforeach; ?>
-<? endif; ?>
+<? endif; ?>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/Recommend/TopFacets.phtml]: I think this should be reverted -- it's removing one of the line endings you fixed a week or two ago.

File: themes/bootprint3/less/common.less [hunk 3]
```diff
@@ -46,6 +46,8 @@ h2 { margin: 8px 0; }
   background: #fff;
 }
 
+h2 { margin: 0 8px 8px; }
```
Comment (demiankatz) [inline themes/bootprint3/less/common.less]: Are the changes to bootprint3 / bootstrap3 less files here actually directly related to the modern theme, or are these either legacy code that should be reverted or new changes that should be separately merged to master?

File: themes/bootstrap3/templates/search/advanced/checkbox-filters.phtml [hunk 4]
```diff
@@ -9,4 +9,4 @@
       </div>
     <? endforeach; ?>
   </div>
-<?endif;?>
+<? endif; ?>
```
Comment (demiankatz) [inline themes/bootstrap3/templates/search/advanced/checkbox-filters.phtml]: This looks like a style fix that should probably be merged to master separately.

File: themes/bootstrap3/templates/myresearch/holds.phtml [hunk 5]
```diff
@@ -98,6 +98,35 @@
               <br />
             <? endif; ?>
 
+            <?
```
Comment (demiankatz) [inline themes/bootstrap3/templates/myresearch/holds.phtml]: What's all this about? There seem to be some related changes in multiple myresearch templates, and it's not clear what any of these have to do with the modern theme. Do you know what's going on here, or is this some kind of merge-related glitch?

File: package.json (hunk 1 above)
Comment (crhallberg) [inline package.json]: This was automatically done by grunt, I can revert it.

File: themes/bootstrap3/templates/Recommend/TopFacets.phtml (hunk 2 above)
Comment (crhallberg) [inline themes/bootstrap3/templates/Recommend/TopFacets.phtml]: All these template changes snuck in on master merge somehow. They all need to get reverted. I don't know why they're here.

**Comment by demiankatz** [comment]
Tabbed search box bug:

image

**Comment by demiankatz** [comment]
Here's the "system under maintenance" bug that I just added a checkbox for above:
image

**Comment by demiankatz** [comment]
Here's the latest bug: tabs + advanced search do not play nicely:

image

**Comment by demiankatz** [comment]
And here's another one -- weird caret in "remove filters" box:
image

**Comment by demiankatz** [comment]
And another one -- contrast problems for selected facets:
image

**Comment by crhallberg** [comment]
All Firefox, correct?

**Comment by demiankatz** [comment]
Correct.

**Pull request merged**